.github/workflows/workflow.yml

```diff
@@ -0,0 +1,48 @@
+name: CI workflow for rebel hub
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    # uwuntu latest for simplicity
+    # python 3.12 container to match our proj
+    name: Build & Run Tests
+    runs-on: ubuntu-latest
+    container: python:3.12-slim
+    steps:
+      # fist we checkout the repo
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      # we install the backend deps listed in requirements.txt
+      - name: Install Backend Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r backend/requirements.txt
+
+      # HERE WE CAN ADD IN STEPS FOR LINTING/SECURITY/AND FORMATTING
+
+      # we run all the backend tests
+      - name: Run Django Backend Tests
+        run: |
+          cd backend
+          python manage.py test
```

Comment:
So we're using the manage.py test for the backend tests instead of pytest? Since we have pytest, pytest-cov, and pytest-django, we could use pytest to make a more detailed report for the backend tests.

Reply:
good call, i'll reconfigure it to use pytest instead with a more detailed report. i'll come back to this when thats done.